crates/sqllib/src/dec.rs

```diff
@@ -0,0 +1,1067 @@
+    }
+
+    fn visit_map<A>(self, map: A) -> Result<SqlDecimal, A::Error>
+    where
+        A: serde::de::MapAccess<'de>,
+    {
+        let mut map = map;
+        let value = map.next_key::<DecimalKey>()?;
+        if value.is_none() {
+            return Err(serde::de::Error::invalid_type(Unexpected::Map, &self));
+        }
+        let v: DecimalFromString = map.next_value()?;
+        Ok(v.value)
+    }
+}
+
+struct DecimalKey;
+const DECIMAL_KEY_TOKEN: &str = "$serde_json::private::Number";
+
+impl<'de> serde::de::Deserialize<'de> for DecimalKey {
+    fn deserialize<D>(deserializer: D) -> Result<DecimalKey, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        struct FieldVisitor;
+
+        impl serde::de::Visitor<'_> for FieldVisitor {
+            type Value = ();
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a valid DECIMAL field")
+            }
+
+            fn visit_str<E>(self, s: &str) -> Result<(), E>
+            where
+                E: serde::de::Error,
+            {
+                if s == DECIMAL_KEY_TOKEN {
+                    Ok(())
+                } else {
+                    Err(serde::de::Error::custom("expected field with custom name"))
+                }
+            }
+        }
+
+        deserializer.deserialize_identifier(FieldVisitor)?;
+        Ok(DecimalKey)
+    }
+}
+
+pub struct DecimalFromString {
+    pub value: SqlDecimal,
+}
+
+impl<'de> serde::de::Deserialize<'de> for DecimalFromString {
+    fn deserialize<D>(deserializer: D) -> Result<DecimalFromString, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        struct Visitor;
+
+        impl serde::de::Visitor<'_> for Visitor {
+            type Value = DecimalFromString;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("string containing a DECIMAL")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<DecimalFromString, E>
+            where
+                E: serde::de::Error,
+            {
+                let d = SqlDecimal::parse(value).map_err(serde::de::Error::custom)?;
+                Ok(DecimalFromString { value: d })
+            }
+        }
+
+        deserializer.deserialize_str(Visitor)
+    }
+}
+
+//////////////////////
+
+impl SerializeWithContext<SqlSerdeConfig> for SqlDecimal {
+    fn serialize_with_context<S>(
+        &self,
+        serializer: S,
+        context: &SqlSerdeConfig,
+    ) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match context.decimal_format {
+            DecimalFormat::Numeric => Serialize::serialize(&self, serializer),
+            DecimalFormat::String => serializer.serialize_str(&self.value.to_string()),
+            DecimalFormat::U128 => todo!(), // serializer.serialize_u128(u128::from_be_bytes(self.serialize())),
```

Comment:
This is used by the Avro encode, need to figure out how to deal with that. 
Please remove the same code for `Decimal` in `crates/feldera-types/src/serde_with_context/serialize.rs`

Reply:
I think I tried to remove the code, but then some tests would not compile anymore.